node/silkworm/db/buffer.cpp

```diff
@@ -158,36 +146,37 @@ void Buffer::write_to_db() {
 
     auto incarnation_table{db::open_cursor(txn_, table::kIncarnationMap)};
     Bytes data(kIncarnationLength, '\0');
-    for (const auto& entry : incarnations_) {
-        endian::store_big_u64(&data[0], entry.second);
-        incarnation_table.upsert(to_slice(entry.first), to_slice(data));
+    for (const auto& [address, incarnation] : incarnations_) {
+        endian::store_big_u64(&data[0], incarnation);
+        incarnation_table.upsert(to_slice(address), to_slice(data));
     }
 
     auto code_table{db::open_cursor(txn_, table::kCode)};
     for (const auto& entry : hash_to_code_) {
         code_table.upsert(to_slice(entry.first), to_slice(entry.second));
     }
 
-    auto code_hash_table{db::open_cursor(txn_, table::kPlainContractCode)};
+    auto code_hash_table{db::open_cursor(txn_, table::kPlainCodeHash)};
     for (const auto& entry : storage_prefix_to_code_hash_) {
         code_hash_table.upsert(to_slice(entry.first), to_slice(entry.second));
     }
 
     auto account_change_table{db::open_cursor(txn_, table::kAccountChangeSet)};
-    Bytes change_key;
-    for (const auto& block_entry : account_changes_) {
-        uint64_t block_num{block_entry.first};
-        change_key = block_key(block_num);
-        for (const auto& account_entry : block_entry.second) {
-            data = ByteView{account_entry.first};
-            data.append(account_entry.second);
-            auto data_slice{to_slice(data)};
-            account_change_table.put(to_slice(change_key), &data_slice, MDBX_APPENDDUP);
+    Bytes change_key(8, '\0');
+    for (const auto& [block_num, account_changes] : block_account_changes_) {
+        endian::store_big_u64(change_key.data(), block_num);
+        for (const auto& [address, storage_encoded] : account_changes) {
+            Bytes change_value(kAddressLength + storage_encoded.length(), '\0');
```

Comment:
Better declare `change_value` outside of the loops and resize it inside to avoid memory reallocation.

Reply:
See [54bce04](https://github.com/torquem-ch/silkworm/pull/553/commits/54bce04d93244532c481fd82b77845d0973e1ed2)
Only one allocation. No resizing ... only slice (view) is adjusted in length